dhall/tests/Dhall/Test/TypeInference.hs

```diff
@@ -41,24 +49,62 @@ successTest prefix = do
                  -- fails due to `Expr`'s 'Eq' instance, which inherits the
                  -- @NaN /= NaN@ inequality from 'Double'.
                  typeInferenceDirectory </> "success/unit/AssertNaN"
+
+               , typeInferenceDirectory </> "success/prelude" -- TODO
```

Comment:
Although I have the updated import resolution to ignore the semantic cache, the test passes only once this patch is applied:

```diff
--- a/tests/type-inference/success/preludeB.dhall
+++ b/tests/type-inference/success/preludeB.dhall
@@ -398,7 +398,7 @@
     }
 , XML :
     { attribute :
-        Text → Text → { mapKey : Text, mapValue : Text }
+        ∀(key : Text) → ∀(value : Text) → { mapKey : Text, mapValue : Text }
     , element :
           ∀ ( elem
             : { attributes :
```

Reply:
... as noted in https://github.com/dhall-lang/dhall-lang/pull/723#issuecomment-530057232, this is not a bug.  This is the correct inferred type when you disable the cache.